
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Downstream alignment and phylogenetics with QIIME 2 &#8212; genome-sampler documentation</title>
    
  <link rel="stylesheet" href="_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="_static/sphinx-book-theme.2d2078699c18a0efb88233928e1cf6ed.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.be0a4a0c39cd630af62a2fcf693f3f06.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://caporasolab.us/genome-sampler/downstream-workflows.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Other topics" href="other.html" />
    <link rel="prev" title="genome-sampler usage tutorial" href="tutorial.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://caporasolab.us/genome-sampler/downstream-workflows.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Downstream alignment and phylogenetics with QIIME 2" />
<meta property="og:description" content="Downstream alignment and phylogenetics with QIIME 2  QIIME 2 contains tools for sequence alignment, alignment filtering, and phylogenetic reconstruction in the " />


<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  
  <h1 class="site-logo" id="site-title">genome-sampler documentation</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   genome-sampler documentation
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Getting started
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="install.html">
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   installation instructions
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="tutorial.html">
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   usage tutorial
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Next steps
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Downstream alignment and phylogenetics with QIIME 2
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="other.html">
   Other topics
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Back matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="zreferences.html">
   List of works cited
  </a>
 </li>
</ul>

</nav>

 <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    
    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/downstream-workflows.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
    
</div>
        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/caporaso-lab/genome-sampler"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/caporaso-lab/genome-sampler/issues/new?title=Issue%20on%20page%20%2Fdownstream-workflows.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/caporaso-lab/genome-sampler/edit/master/docs/downstream-workflows.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#obtain-reference-sequence-and-alignment-mask">
   Obtain reference sequence and alignment mask
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#align-sequences-and-build-a-tree">
   Align sequences and build a tree
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="downstream-alignment-and-phylogenetics-with-qiime-2">
<span id="downstream"></span><h1>Downstream alignment and phylogenetics with QIIME 2<a class="headerlink" href="#downstream-alignment-and-phylogenetics-with-qiime-2" title="Permalink to this headline">¶</a></h1>
<p>QIIME 2 contains tools for sequence alignment, alignment filtering, and
phylogenetic reconstruction in the
<a class="reference external" href="https://docs.qiime2.org/2020.2/plugins/available/alignment/">q2-alignment</a>
and
<a class="reference external" href="https://docs.qiime2.org/2020.2/plugins/available/phylogeny/">q2-phylogeny</a>
plugins (which are not installed by default with genome-sampler). This document
illustrates how these steps can be applied after working through
<a class="reference internal" href="tutorial.html#usage-tutorial"><span class="std std-ref">genome-sampler usage tutorial</span></a>.</p>
<p>QIIME 2 often wraps other widely used tools in QIIME 2 plugins rather than
implement them directly. For example, under-the-hood, <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code>’s
<code class="docutils literal notranslate"><span class="pre">sample-diversity</span></code> action is using <code class="docutils literal notranslate"><span class="pre">vsearch</span></code> <a class="bibtex reference internal" href="zreferences.html#vsearch-peerj" id="id1">[1]</a>. In this
document we’ll build an alignment with MAFFT <a class="bibtex reference internal" href="zreferences.html#mafft7" id="id2">[2]</a>,
apply a pre-computed alignment position mask, and then build a phylogenetic
tree with IQTree 2 <a class="bibtex reference internal" href="zreferences.html#iqtree2" id="id3">[3]</a>.</p>
<div class="admonition warning">
<p class="admonition-title">Warning</p>
<p>QIIME 2 is not designed for high-accuracy phylogenetic analysis. If you use the
workflow illustrated here, we recommend that you consider the resulting trees
to be experimental. In other words, this workflow can give you a quick idea of
what is happening with your data, but you may want to use other tools for your
final trees.</p>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>This tutorial assumes that you have already run <a class="reference internal" href="tutorial.html#usage-tutorial"><span class="std std-ref">genome-sampler usage tutorial</span></a>. If you
haven’t, you’ll be missing files that are needed for this tutorial.</p>
</div>
<div class="section" id="obtain-reference-sequence-and-alignment-mask">
<h2>Obtain reference sequence and alignment mask<a class="headerlink" href="#obtain-reference-sequence-and-alignment-mask" title="Permalink to this headline">¶</a></h2>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">alignment</span><span class="o">-</span><span class="n">mask</span><span class="o">.</span><span class="n">qza</span> <span class="o">--</span><span class="n">output</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">alignment</span><span class="o">-</span><span class="n">mask</span><span class="o">.</span><span class="n">qza</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">sarscov2</span><span class="o">-</span><span class="n">reference</span><span class="o">-</span><span class="n">genome</span><span class="o">.</span><span class="n">qza</span> <span class="o">--</span><span class="n">output</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">sarscov2</span><span class="o">-</span><span class="n">reference</span><span class="o">-</span><span class="n">genome</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
</div>
<div class="section" id="align-sequences-and-build-a-tree">
<h2>Align sequences and build a tree<a class="headerlink" href="#align-sequences-and-build-a-tree" title="Permalink to this headline">¶</a></h2>
<p>First, we’ll add the SARS-CoV-2 reference sequence to the sequence collection
obtained in the tutorial. Notice that we’re working with the <code class="docutils literal notranslate"><span class="pre">.qza</span></code> file that
was created in that tutorial, not the <code class="docutils literal notranslate"><span class="pre">.fasta</span></code> file that we exported.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">merge</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">sequences</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">sarscov2</span><span class="o">-</span><span class="n">reference</span><span class="o">-</span><span class="n">genome</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">merged</span><span class="o">-</span><span class="n">data</span> <span class="n">sequences</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Next, we’ll perform sequence alignment using MAFFT.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">alignment</span> <span class="n">mafft</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">sequences</span> <span class="n">sequences</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">alignment</span> <span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>After aligning the sequences, a “mask” can be applied to filter positions from
the alignment that are likely to be uninformative. At present, we’re
experimenting with a <a class="reference internal" href="other.html#alignment-mask"><span class="std std-ref">pre-computed alignment mask</span></a>.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">mask</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">alignment</span> <span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">mask</span> <span class="n">alignment</span><span class="o">-</span><span class="n">mask</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">masked</span><span class="o">-</span><span class="n">alignment</span> <span class="n">masked</span><span class="o">-</span><span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Finally, we build a tree from the resulting alignment. This will generate an
unrooted phylogenetic tree.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">phylogeny</span> <span class="n">iqtree</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">alignment</span> <span class="n">masked</span><span class="o">-</span><span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">ref</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">tree</span> <span class="n">unrooted</span><span class="o">-</span><span class="n">tree</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>This <code class="docutils literal notranslate"><span class="pre">.qza</span></code> file can
be viewed directly with <a class="reference external" href="https://itol.embl.de/">iTOL</a> to get a quick look.
We’ll also soon be adding support for viewing this with
<a class="reference external" href="https://github.com/biocore/empress">Empress</a>.</p>
<p>All of the <code class="docutils literal notranslate"><span class="pre">.qza</span></code> files that were generated in this example can be exported
using <code class="docutils literal notranslate"><span class="pre">qiime</span> <span class="pre">tools</span> <span class="pre">export</span></code>. Exporting of sequence or alignment files will
provide you with fasta files by default, and exporting of the phylogenetic
tree will provide you with a newick file by default. See the
<a class="reference external" href="https://docs.qiime2.org/2020.8/tutorials/exporting/">QIIME 2 exporting documentation</a>
for more details.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="tutorial.html" title="previous page"><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> usage tutorial</a>
    <a class='right-next' id="next-link" href="other.html" title="next page">Other topics</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The genome-sampler developers<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="_static/js/index.3da636dd464baa7582d2.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-86713402-2', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>