
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>Other topics &#8212; genome-sampler documentation</title>
    
  <link rel="stylesheet" href="_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="_static/sphinx-book-theme.2d2078699c18a0efb88233928e1cf6ed.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.be0a4a0c39cd630af62a2fcf693f3f06.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://caporasolab.us/genome-sampler/other.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="List of works cited" href="zreferences.html" />
    <link rel="prev" title="Downstream alignment and phylogenetics with QIIME 2" href="downstream-workflows.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://caporasolab.us/genome-sampler/other.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="Other topics" />
<meta property="og:description" content="Other topics    Sampling focal sequences  genome-sampler can be used for sampling focal sequences in addition to context sequences. In fact, all of the exact sa" />


<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  
  <h1 class="site-logo" id="site-title">genome-sampler documentation</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   genome-sampler documentation
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Getting started
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="install.html">
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   installation instructions
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="tutorial.html">
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   usage tutorial
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Next steps
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="downstream-workflows.html">
   Downstream alignment and phylogenetics with QIIME 2
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   Other topics
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Back matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="zreferences.html">
   List of works cited
  </a>
 </li>
</ul>

</nav>

 <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    
    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/other.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
    
</div>
        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/caporaso-lab/genome-sampler"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/caporaso-lab/genome-sampler/issues/new?title=Issue%20on%20page%20%2Fother.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/caporaso-lab/genome-sampler/edit/master/docs/other.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#sampling-focal-sequences">
   Sampling focal sequences
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#running-in-parallel">
   Running in parallel
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#adapting-snakemake-workflow-for-application-to-your-own-data">
   Adapting Snakemake workflow for application to your own data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#pre-computed-alignment-mask">
   Pre-computed alignment mask
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#removing-sequences-present-in-both-focal-and-context-sequence-collections">
   Removing sequences present in both focal and context sequence collections
  </a>
  <ul class="nav section-nav flex-column">
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#shared-identifiers">
     Shared identifiers
    </a>
   </li>
   <li class="toc-h3 nav-item toc-entry">
    <a class="reference internal nav-link" href="#differing-identifiers">
     Differing identifiers
    </a>
   </li>
  </ul>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="other-topics">
<span id="other"></span><h1>Other topics<a class="headerlink" href="#other-topics" title="Permalink to this headline">¶</a></h1>
<div class="section" id="sampling-focal-sequences">
<span id="id1"></span><h2>Sampling focal sequences<a class="headerlink" href="#sampling-focal-sequences" title="Permalink to this headline">¶</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> can be used for sampling focal sequences in addition to
context sequences. In fact, all of the exact same sampling approaches can
be applied in exactly the same way. The only additional file you’ll need to do
this (relative to the files used in <a class="reference internal" href="tutorial.html#usage-tutorial"><span class="std std-ref">genome-sampler usage tutorial</span></a>) is a focal sequence
metadata file. This can mirror the format of the context sequence metadata
file. For more details on the format of these files, see
<a class="reference external" href="https://docs.qiime2.org/2020.6/tutorials/metadata/">Metadata in QIIME 2</a>.</p>
</div>
<div class="section" id="running-in-parallel">
<span id="parallel"></span><h2>Running in parallel<a class="headerlink" href="#running-in-parallel" title="Permalink to this headline">¶</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> can be run in parallel to speed it up. This is done in
different ways depending on whether you’re running the steps individually or
through Snakemake.</p>
<p>If you’re using Snakemake, you need only pass additional cores (via
<code class="docutils literal notranslate"><span class="pre">snakemake</span> <span class="pre">--cores</span> <span class="pre">N</span></code>). The parallelizable steps will automatically use the
provided resources.</p>
<p>If you’re running the steps individually you can pass the <code class="docutils literal notranslate"><span class="pre">--p-n-threads</span></code>
option to several of the commands. For example, <code class="docutils literal notranslate"><span class="pre">sample-diversity</span></code> is the
slowest step in the workflow. You can provide the <code class="docutils literal notranslate"><span class="pre">--p-n-threads</span></code> parameter to
run it in parallel:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">diversity</span> \
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">percent</span><span class="o">-</span><span class="nb">id</span> <span class="mf">0.9995</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">n</span><span class="o">-</span><span class="n">threads</span> <span class="n">N</span>
</pre></div>
</div>
<p>When running this command, you should set <code class="docutils literal notranslate"><span class="pre">N</span></code> to be the number of available
processors or cores on a single node of your system. For example, on a
cluster that has nodes with 28 cores, you could run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">diversity</span> \
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">percent</span><span class="o">-</span><span class="nb">id</span> <span class="mf">0.9995</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">n</span><span class="o">-</span><span class="n">threads</span> <span class="mi">28</span>
</pre></div>
</div>
<p>This would use all of the resources on a single node of that cluster. In the
future QIIME 2 will have support for splitting workflows like this across
multiple cluster nodes, but we do not have this support now.</p>
</div>
<div class="section" id="adapting-snakemake-workflow-for-application-to-your-own-data">
<span id="adapting-tutorial"></span><h2>Adapting Snakemake workflow for application to your own data<a class="headerlink" href="#adapting-snakemake-workflow-for-application-to-your-own-data" title="Permalink to this headline">¶</a></h2>
<p>The Snakemake workflow presented in the tutorial is a good starting point for
your own analyses. There are typically a few things to do to adapt
<code class="docutils literal notranslate"><span class="pre">Snakefile</span></code> for your own data. These changes will be made to the <code class="docutils literal notranslate"><span class="pre">config.yaml</span></code>
file which should be found alongside the <code class="docutils literal notranslate"><span class="pre">Snakefile</span></code>.</p>
<ol class="simple">
<li><p>Modify input filepaths as needed. The input filepaths listed correspond to
the names of the files provided for the tutorial. You can either name your
files using those names, or update the input filepath values.</p></li>
<li><p>Modify output filepaths if you’d like these to be different from the ones
used in the tutorial.</p></li>
<li><p>Modify longitudinal, neighbor, and diversity sampling parameters as
desired. If you end up experimenting with different values for these
parameters, which we encourage, we would love to hear about your findings. Be
aware that increasing the <code class="docutils literal notranslate"><span class="pre">*_percent_id</span></code> parameters will increase the runtime
of your analysis, and decreasing those values will decrease the runtime of your
analysis.</p></li>
</ol>
</div>
<div class="section" id="pre-computed-alignment-mask">
<span id="alignment-mask"></span><h2>Pre-computed alignment mask<a class="headerlink" href="#pre-computed-alignment-mask" title="Permalink to this headline">¶</a></h2>
<p>We provide support for masking alignments with the pre-computed mask
provided <a class="reference external" href="https://github.com/W-L/ProblematicSites_SARS-CoV2">here</a>. Like the
rest of our <a class="reference internal" href="downstream-workflows.html#downstream"><span class="std std-ref">alignment and phylogenetics workflow</span></a>,
this is experimental. Let us know how it works for you. It would be very useful
to hear about how your final results (e.g., a phylogenetic tree) computed
using this mask compares to one created with your own custom mask.</p>
</div>
<div class="section" id="removing-sequences-present-in-both-focal-and-context-sequence-collections">
<span id="removing-sequences"></span><h2>Removing sequences present in both focal and context sequence collections<a class="headerlink" href="#removing-sequences-present-in-both-focal-and-context-sequence-collections" title="Permalink to this headline">¶</a></h2>
<p>As many researchers are rapidly submitting their sequence data to public
repositories, sometimes your focal sequences may be present in context
sequences that you obtain from a public repository. You can use QIIME 2 to
remove those sequences from your context sequence collection.</p>
<p>First, visualize your focal sequence and context sequence collections with
QIIME 2 to confirm that you know what QIIME 2 considers to be the sequence
identifiers. You can do this by running the following two commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">tabulate</span><span class="o">-</span><span class="n">seqs</span> \
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qzv</span>

<span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">tabulate</span><span class="o">-</span><span class="n">seqs</span> \
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qzv</span>
</pre></div>
</div>
<p>Then, load the two visualizations with <a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>.
The values in the <em>Feature ID</em> column are the sequence ids. If the ids of the
sequences that you want to remove are the same across your two data sets, you
can follow the steps under <a class="reference internal" href="#shared-identifiers"><span class="std std-ref">Shared identifiers</span></a>. If the identifiers you
want to remove are not the same across your two data sets, or you’re not sure,
follow the steps under <a class="reference internal" href="#differing-identifiers"><span class="std std-ref">Differing identifiers</span></a>.</p>
<div class="section" id="shared-identifiers">
<span id="id2"></span><h3>Shared identifiers<a class="headerlink" href="#shared-identifiers" title="Permalink to this headline">¶</a></h3>
<p>If your sequence identifiers are identical across your focal and context
sequence collections, you can remove context sequences that are also focal
sequences as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span>\
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="n">focal</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">exclude</span><span class="o">-</span><span class="n">ids</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>All of your downstream analysis steps should then use the resulting
<code class="docutils literal notranslate"><span class="pre">filtered-context-seqs.qza</span></code> file.</p>
<p>Alternatively, if you want to remove these sequences from your focal sequence
collection, you could do this as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span>\
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">exclude</span><span class="o">-</span><span class="n">ids</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
</div>
<div class="section" id="differing-identifiers">
<span id="id3"></span><h3>Differing identifiers<a class="headerlink" href="#differing-identifiers" title="Permalink to this headline">¶</a></h3>
<p>If your sequence identifiers differ across your focal and context sequences (or
you’re not sure if they’re the same or if they differ), you can perform a
metadata-based filtering of your sequences. The easiest way to approach this
is to add a new column to your context sequence metadata file
(<code class="docutils literal notranslate"><span class="pre">context-metadata.tsv</span></code> in the tutorial), with a column header like
<code class="docutils literal notranslate"><span class="pre">is-focal-sequence</span></code>. You can do this with your
favorite spreadsheet editor. For all of the sequences you would like to remove,
add the value <code class="docutils literal notranslate"><span class="pre">TRUE</span></code> to this column. For all of the sequences you would like to
retain, add the value <code class="docutils literal notranslate"><span class="pre">FALSE</span></code> to this column. Then, save the file and run the
following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
 <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
 <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
 <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">where</span> <span class="s2">&quot;[is_focal_sequence]=&#39;FALSE&#39;&quot;</span> \
 <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>This will create a new file, <code class="docutils literal notranslate"><span class="pre">filtered-context-seqs.qza</span></code>, which only contains
the sequences that you indicated were not focal sequences in the context
metadata. All of your downstream analysis steps should then use the resulting
<code class="docutils literal notranslate"><span class="pre">filtered-context-seqs.qza</span></code> file.</p>
<p>Alternatively, if you want to remove sequences from your focal sequence
collection instead of from your context sequence collection, you can edit your
focal sequence metadata file, and then use that to filter in the same way that
context sequences were filtered here.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If your metadata file already contains a column that differentiates the
sequences you want to remove from the sequences you want to retain, it’s not
necessary to add a new column. Instead, you can design a value for the
<code class="docutils literal notranslate"><span class="pre">--p-where</span></code> parameter that will allow the filter to operate on data from
that column. For more information on metadata-based filtering, see
<a class="reference external" href="https://docs.qiime2.org/2020.8/tutorials/filtering/#metadata-based-filtering">Metadata-based filtering</a>
in the QIIME 2 documentation.</p>
</div>
</div>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="downstream-workflows.html" title="previous page">Downstream alignment and phylogenetics with QIIME 2</a>
    <a class='right-next' id="next-link" href="zreferences.html" title="next page">List of works cited</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The genome-sampler developers<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="_static/js/index.3da636dd464baa7582d2.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-86713402-2', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>