

<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>genome-sampler installation and usage tutorial &#8212; My Jupyter Book</title>
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css" integrity="sha384-KA6wR/X5RY4zFAHpv/CnoG2UW1uogYfdnP67Uv7eULvTveboZJg0qUpmJZb5VqzN" crossorigin="anonymous">
    <link href="_static/css/index.css" rel="stylesheet">
    <link rel="stylesheet" href="_static/sphinx-book-theme.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/jupyter-sphinx.css" />
    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/sphinx-book-theme.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script src="_static/mystnb.js"></script>
    <script src="_static/sphinx-book-theme.js"></script>
    <script >var togglebuttonSelector = '.toggle, .secondtoggle, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/require.js/2.3.4/require.min.js"></script>
    <script src="https://unpkg.com/@jupyter-widgets/html-manager@^0.18.0/dist/embed-amd.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="docsearch:language" content="en">



  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  
  <h1 class="site-logo" id="site-title">My Jupyter Book</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  
  
</nav>

 <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse" data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu" aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation" title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
            <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i class="fas fa-download"></i></button>

            
            <div class="dropdown-buttons">
                <!-- ipynb file if we had a myst markdown file -->
                
                <!-- Download raw file -->
                <a class="dropdown-buttons" href="_sources/tutorial.md.txt"><button type="button" class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip" data-placement="left">.md</button></a>
                <!-- Download PDF via print -->
                <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF" onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
            </div>
            
        </div>

        <!-- Source interaction buttons -->
        
        <div class="dropdown-buttons-trigger">
            <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
            <div class="dropdown-buttons sourcebuttons">
                <a class="repository-button" href="https://github.com/executablebooks/jupyter-book"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left" title="Source repository"><i class="fab fa-github"></i>repository</button></a>
                <a class="issues-button" href="https://github.com/executablebooks/jupyter-book/issues/new?title=Issue%20on%20page%20%2Ftutorial.html&body=Your%20issue%20content%20here."><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left" title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
                
            </div>
        </div>
        

        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="bottom" onclick="toggleFullScreen()" title="Fullscreen mode"><i class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->
        
    </div>
    <div class="d-none d-md-block col-md-2 bd-toc show">
<div class="tocsection onthispage pt-5 pb-3">
    <i class="fas fa-list"></i> On this page
</div>

<nav id="bd-toc-nav">
    <ul class="nav section-nav flex-column">
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#installation-instructions" class="nav-link">Installation instructions</a><ul class="nav section-nav flex-column">
                
        <li class="nav-item toc-entry toc-h3">
            <a href="#install-miniconda" class="nav-link">Install Miniconda</a>
        </li>
    
        <li class="nav-item toc-entry toc-h3">
            <a href="#install-genome-sampler-from-source" class="nav-link">Install genome-sampler from source</a>
        </li>
    
            </ul>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#usage-instructions" class="nav-link">Usage instructions</a><ul class="nav section-nav flex-column">
                
        <li class="nav-item toc-entry toc-h3">
            <a href="#download-tutorial-data" class="nav-link">Download tutorial data</a>
        </li>
    
        <li class="nav-item toc-entry toc-h3">
            <a href="#using-genome-sampler-snakemake-workflow" class="nav-link">Using genome-sampler (Snakemake workflow)</a>
        </li>
    
        <li class="nav-item toc-entry toc-h3">
            <a href="#using-genome-sampler-step-by-step-instructions" class="nav-link">Using genome-sampler (step-by-step instructions)</a>
        </li>
    
            </ul>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#getting-help-contributing-etc" class="nav-link">Getting help, contributing, etc</a>
        </li>
    
        <li class="nav-item toc-entry toc-h2">
            <a href="#citing-genome-sampler" class="nav-link">Citing genome-sampler</a>
        </li>
    
    </ul>
</nav>


    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="genome-sampler-installation-and-usage-tutorial">
<h1><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> installation and usage tutorial<a class="headerlink" href="#genome-sampler-installation-and-usage-tutorial" title="Permalink to this headline">¶</a></h1>
<p>This document provides instructions for installing and using genome-sampler.</p>
<div class="section" id="installation-instructions">
<h2>Installation instructions<a class="headerlink" href="#installation-instructions" title="Permalink to this headline">¶</a></h2>
<div class="section" id="install-miniconda">
<h3>Install Miniconda<a class="headerlink" href="#install-miniconda" title="Permalink to this headline">¶</a></h3>
<p><a class="reference external" href="https://conda.io/miniconda.html">Miniconda</a> provides the conda environment
and package manager, and is the recommended way to install <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code>.
Follow the instructions for downloading and installing Miniconda. You may
choose either Miniconda2 or Miniconda3 (i.e. Miniconda Python 2 or 3).
<code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> will work with either version of Miniconda.</p>
</div>
<div class="section" id="install-genome-sampler-from-source">
<h3>Install <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> from source<a class="headerlink" href="#install-genome-sampler-from-source" title="Permalink to this headline">¶</a></h3>
<p>A conda package will be available in the near future. For the moment we only
a source installation.</p>
<p>First create a suitable conda environment:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">create</span> <span class="o">-</span><span class="n">y</span> <span class="o">-</span><span class="n">n</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span>
<span class="n">conda</span> <span class="n">activate</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span>
</pre></div>
</div>
<p>Next install dependencies:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> \
  <span class="o">-</span><span class="n">c</span> <span class="n">conda</span><span class="o">-</span><span class="n">forge</span> <span class="o">-</span><span class="n">c</span> <span class="n">bioconda</span> <span class="o">-</span><span class="n">c</span> <span class="n">qiime2</span> <span class="o">-</span><span class="n">c</span> <span class="n">defaults</span> \
  <span class="n">qiime2</span> <span class="n">q2cli</span> <span class="n">q2templates</span> <span class="n">q2</span><span class="o">-</span><span class="n">types</span> <span class="n">q2</span><span class="o">-</span><span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">q2</span><span class="o">-</span><span class="n">metadata</span> <span class="n">vsearch</span> <span class="n">snakemake</span>
</pre></div>
</div>
<p>Finally install from source:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">git</span><span class="o">+</span><span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">github</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">.</span><span class="n">git</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="usage-instructions">
<h2>Usage instructions<a class="headerlink" href="#usage-instructions" title="Permalink to this headline">¶</a></h2>
<div class="section" id="download-tutorial-data">
<h3>Download tutorial data<a class="headerlink" href="#download-tutorial-data" title="Permalink to this headline">¶</a></h3>
<p>The tutorial data set used here is intended for educational purposes only. If
you’re interested in using these sequences for other analyses, we recommend
starting with sequence repositories such as GISAID or NCBI Genbank, which may
have updated versions.</p>
<p>Download metadata as tsv from the two tabs in <a class="reference external" href="https://docs.google.com/spreadsheets/d/18IyZK6gvwcqKrl2U1FnucrC71Q5VSy_qFTz4ktffNK4/edit#gid=0">this
spreadsheet</a>.
Name the file downloaded from the <code class="docutils literal notranslate"><span class="pre">context-metadata</span></code> tab
<code class="docutils literal notranslate"><span class="pre">context-metadata.tsv</span></code> and the file downloaded from the <code class="docutils literal notranslate"><span class="pre">focal-metadata</span></code> tab
<code class="docutils literal notranslate"><span class="pre">focal-metadata.tsv</span></code>.</p>
<p>Download the two fasta files from <a class="reference external" href="https://www.dropbox.com/sh/tkb0c4snk5zodj8/AABLCykSiEe5zqv8gTeOSegna?dl=0">this Dropbox
folder</a>.</p>
</div>
<div class="section" id="using-genome-sampler-snakemake-workflow">
<h3>Using <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> (Snakemake workflow)<a class="headerlink" href="#using-genome-sampler-snakemake-workflow" title="Permalink to this headline">¶</a></h3>
<p>The full <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> workflow can be run using Snakemake. If you’d like
to get started quickly and use default parameters, start here. If you’d like
more control over your analysis or want to work through the steps
individually, move on to the next section.</p>
<p>Download the Snakemake file using <code class="docutils literal notranslate"><span class="pre">curl</span></code> as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">master</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">Snakefile</span> <span class="o">--</span><span class="n">output</span> <span class="n">Snakefile</span>
</pre></div>
</div>
<p>Place the resulting Snakemake file in the same folder as the sequence and
metadata files that you downloaded above. Then run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span>
</pre></div>
</div>
<p>When this workflow completes, there will be two primary output files that
you’ll use. <code class="docutils literal notranslate"><span class="pre">sequences.fasta</span></code> will contain your subsampled context sequences
and your focal sequences. You should use this file for downstream analyses,
such as alignment and phylogenetic analyses. <code class="docutils literal notranslate"><span class="pre">selection-summary.qzv</span></code> will
provide a summary of the sampling run. You can view this file using <a class="reference external" href="https://view.qiime2.org">QIIME 2
View</a>. Be sure to click the <em>Provenance</em> tab after
loading the file on that page - that will provide full details on the
workflow that was executed.</p>
<p>You should now be able to move on to analysis of your own data. If you’d like
to modify parameters of the workflow, you can do so by opening the Snakemake
file in a text editor and editing the values in the <code class="docutils literal notranslate"><span class="pre">CONFIGS</span></code> section.</p>
<p>🐍</p>
</div>
<div class="section" id="using-genome-sampler-step-by-step-instructions">
<h3>Using <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> (step-by-step instructions)<a class="headerlink" href="#using-genome-sampler-step-by-step-instructions" title="Permalink to this headline">¶</a></h3>
<p>You’ll begin the workflow by importing the fasta files into QIIME 2
Artifacts. QIIME 2 Artifacts are structured zip files which will contain the
data (still in fasta format), but also some QIIME 2-specific information that
(among other things) automates your method recording so you don’t have to do
that.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">tools</span> <span class="kn">import</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">path</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">path</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">format</span> <span class="n">GISAIDDNAFASTAFormat</span> \
  <span class="o">--</span><span class="nb">type</span> <span class="n">FeatureData</span><span class="p">[</span><span class="n">Sequence</span><span class="p">]</span>

<span class="n">qiime</span> <span class="n">tools</span> <span class="kn">import</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">path</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">path</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">format</span> <span class="n">GISAIDDNAFASTAFormat</span> \
  <span class="o">--</span><span class="nb">type</span> <span class="n">FeatureData</span><span class="p">[</span><span class="n">Sequence</span><span class="p">]</span>
</pre></div>
</div>
<p>Next, we’ll apply a quality filter to the sequence data. Technically this is
an optional step for both the context and focal sequences, but in practice if
you obtain your context sequences from a public repository you should
consider the context sequence filtering to be essential. You can use this to
remove sequences based on their length (if they’re too short or too long),
and based on the proportion of ambiguous (e.g., <code class="docutils literal notranslate"><span class="pre">N</span></code>) nucleotide characters
that they contain. Here we’ll apply this both to the context sequences and
the focal sequences. We’ll remove sequences with a proportion of 1% or more
ambiguous characters.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">sequences</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="nb">max</span><span class="o">-</span><span class="n">proportion</span><span class="o">-</span><span class="n">ambiguous</span> <span class="mf">0.01</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">sequences</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">sequences</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="nb">max</span><span class="o">-</span><span class="n">proportion</span><span class="o">-</span><span class="n">ambiguous</span> <span class="mf">0.01</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">sequences</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>If you’d like to see what other options you can control during this filtering
step, you can run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">--help</span></code> parameter can be provided to any of the commands that are used
in this tutorial.</p>
<p>At any time, you could get some summary information about your sequences
using the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">tabulate</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>That command will create a QIIME 2 visualization that you can view using
<a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>. Try viewing that file and finding
information such as the number of sequences present in this file and the
median length of the sequences. (Your data is not uploaded to a server when
you visit <a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>, so you don’t need to be
conerned about exposing sensitive research data.)</p>
<p>We’re now ready to start sampling our data, and we’ll do this in three steps.
These subsampling steps are independent, so can be run in any order.</p>
<p>First, we’ll sample across time. By default, this will select seven genomes
per seven day period. The file that gets generated as a result here isn’t
something that is very useful to view directly, but we’ll use it in a few
minutes to see how many sequences were retained at this step.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">longitudinal</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">dates</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">dates</span><span class="o">-</span><span class="n">column</span> <span class="n">date</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">date</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Next, we’ll sample across viral diversity. This will cluster sequences at a
percent identity threshold of 99.95%, and select the centroid sequence from
each cluster to include in downstream analyses.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">diversity</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">percent</span><span class="o">-</span><span class="nb">id</span> <span class="mf">0.9995</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Last, we’ll sample near neighbors of the focal sequences from the context
sequences. Of the near-neighbor sequences that are identified for each focal
sequence (up to 10 by default), 3 sequences will be selected at random such
that each <code class="docutils literal notranslate"><span class="pre">location</span></code> (defined in the <code class="docutils literal notranslate"><span class="pre">context-metadata.tsv</span></code> file) has an
equal probability of being selected for inclusion in downstream analysis.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">neighbors</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">locale</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">locale</span><span class="o">-</span><span class="n">column</span> <span class="n">location</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">percent</span><span class="o">-</span><span class="nb">id</span> <span class="mf">0.9999</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">samples</span><span class="o">-</span><span class="n">per</span><span class="o">-</span><span class="n">cluster</span> <span class="mi">3</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">neighbor</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Now, we’ll combine the results of the three sampling approaches and generate
a summary of the full selection process.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">combine</span><span class="o">-</span><span class="n">selections</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">selections</span> <span class="n">date</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">neighbor</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">combined</span><span class="o">-</span><span class="n">selection</span> <span class="n">master</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">summarize</span><span class="o">-</span><span class="n">selections</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">selections</span> <span class="n">date</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">neighbor</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="n">selection</span><span class="o">-</span><span class="n">summary</span><span class="o">.</span><span class="n">qzv</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">selection-summary.qzv</span></code> file provides a summary of the sampling run. You
can view this file using <a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>. Be sure to
click the <em>Provenance</em> tab after loading the file on that page - that will
provide full details on the workflow that was executed. How many sequences
were retained by each sampling step?</p>
<p>We’re now ready to start compiling our final data set. To do this, we’ll use
the <code class="docutils literal notranslate"><span class="pre">master-selection.qza</span></code> file to select specific context sequences from the
<code class="docutils literal notranslate"><span class="pre">filtered-context-seqs.qza</span></code> file that was generated earlier.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="n">master</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">data</span> <span class="n">subsampled</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Next, merge the resulting subsampled context sequences with the focal
sequences to create the final QIIME 2 artifact. Then, export that to generate
a fasta file that you can use in downstream analysis.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">merge</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">subsampled</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">merged</span><span class="o">-</span><span class="n">data</span> <span class="n">sequences</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">tools</span> <span class="n">export</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">path</span> <span class="n">sequences</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">path</span> <span class="n">sequences</span><span class="o">.</span><span class="n">fasta</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="nb">format</span> <span class="n">DNAFASTAFormat</span>
</pre></div>
</div>
<p><strong>Optional</strong>: QIIME 2 contains some tools for sequence alignment and
phylogenetic reconstruction in the
<a class="reference external" href="https://docs.qiime2.org/2020.2/plugins/available/alignment/">q2-alignment</a>
and
<a class="reference external" href="https://docs.qiime2.org/2020.2/plugins/available/phylogeny/">q2-phylogeny</a>
plugins. If you’d like, you can use these for the next steps of your
analyses. These would take the <code class="docutils literal notranslate"><span class="pre">sequences.qza</span></code> file as input, so you could
just postpone the export step that you ran above. For example, you could
align and build a tree as follows. Note however that usually you would
perform some manual filtering and trimming between these two steps, so these
two commands likely won’t get you a publication quality phylogeny.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">alignment</span> <span class="n">mafft</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">sequences</span> <span class="n">sequences</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span> <span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">phylogeny</span> <span class="n">fasttree</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">alignment</span> <span class="n">aligned</span><span class="o">-</span><span class="n">sequences</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">tree</span> <span class="n">tree</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="getting-help-contributing-etc">
<h2>Getting help, contributing, etc<a class="headerlink" href="#getting-help-contributing-etc" title="Permalink to this headline">¶</a></h2>
<p><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> is open source and free for all use. Software and unit tests
are available at https://github.com/caporaso-lab/genome-sampler under the BSD
3-clause license. Documentation, written using
<a class="reference external" href="https://myst-parser.readthedocs.io/en/latest">Myst</a> and rendered using
<a class="reference external" href="https://jupyterbook.org/">Jupyter Book</a>, is available at
http://caporasolab.us/genome-sampler/.</p>
<p>If you need technical support, please post a question to the QIIME 2 Forum at
https://forum.qiime2.org. We are very interested in contributions to
genome-sampler from the community - please get in touch via the GitHub issue
tracker or the QIIME 2 Forum if you’re interested in contributing. Before
getting in touch, please review the software project’s <a class="reference external" href="https://github.com/caporaso-lab/code-of-conduct/blob/master/code-of-conduct.md">code of
conduct</a>,
which is adapted from the <a class="reference external" href="https://www.contributor-covenant.org">Contributor
Covenant</a>, version 1.4.</p>
</div>
<div class="section" id="citing-genome-sampler">
<h2>Citing <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code><a class="headerlink" href="#citing-genome-sampler" title="Permalink to this headline">¶</a></h2>
<p>If you use <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> in published work, please cite our pre-print (link to follow when available).</p>
</div>
</div>


              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    <script src="_static/js/index.js"></script>
    
  </body>
</html>