
<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta charset="utf-8" />
    <title>genome-sampler usage tutorial &#8212; genome-sampler documentation</title>
    
  <link rel="stylesheet" href="_static/css/index.73d71520a4ca3b99cfee5594769eaaae.css">

    
  <link rel="stylesheet"
    href="_static/vendor/fontawesome/5.13.0/css/all.min.css">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-solid-900.woff2">
  <link rel="preload" as="font" type="font/woff2" crossorigin
    href="_static/vendor/fontawesome/5.13.0/webfonts/fa-brands-400.woff2">

    
      
  <link rel="stylesheet"
    href="_static/vendor/open-sans_all/1.44.1/index.css">
  <link rel="stylesheet"
    href="_static/vendor/lato_latin-ext/1.44.1/index.css">

    
    <link rel="stylesheet" href="_static/sphinx-book-theme.2d2078699c18a0efb88233928e1cf6ed.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="_static/togglebutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/copybutton.css" />
    <link rel="stylesheet" type="text/css" href="_static/mystnb.css" />
    <link rel="stylesheet" type="text/css" href="_static/sphinx-thebe.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-main.c949a650a448cc0ae9fd3441c0e17fb0.css" />
    <link rel="stylesheet" type="text/css" href="_static/panels-variables.06eb56fa6e07937060861dad626602ad.css" />
    
  <link rel="preload" as="script" href="_static/js/index.3da636dd464baa7582d2.js">

    <script id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <script src="_static/language_data.js"></script>
    <script src="_static/togglebutton.js"></script>
    <script src="_static/clipboard.min.js"></script>
    <script src="_static/copybutton.js"></script>
    <script >var togglebuttonSelector = '.toggle, .admonition.dropdown, .tag_hide_input div.cell_input, .tag_hide-input div.cell_input, .tag_hide_output div.cell_output, .tag_hide-output div.cell_output, .tag_hide_cell.cell, .tag_hide-cell.cell';</script>
    <script src="_static/sphinx-book-theme.be0a4a0c39cd630af62a2fcf693f3f06.js"></script>
    <script async="async" src="https://unpkg.com/thebelab@latest/lib/index.js"></script>
    <script >
        const thebe_selector = ".thebe"
        const thebe_selector_input = "pre"
        const thebe_selector_output = ".output"
    </script>
    <script async="async" src="_static/sphinx-thebe.js"></script>
    <link rel="canonical" href="https://caporasolab.us/genome-sampler/tutorial.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Downstream alignment and phylogenetics with QIIME 2" href="downstream-workflows.html" />
    <link rel="prev" title="genome-sampler installation instructions" href="install.html" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="docsearch:language" content="en" />


<!-- Opengraph tags -->
<meta property="og:url"         content="https://caporasolab.us/genome-sampler/tutorial.html" />
<meta property="og:type"        content="article" />
<meta property="og:title"       content="genome-sampler usage tutorial" />
<meta property="og:description" content="genome-sampler usage tutorial  This document provides illustrates how to use genome-sampler on a small tutorial data set.  Download tutorial data  The tutorial " />


<meta name="twitter:card" content="summary" />


  </head>
  <body data-spy="scroll" data-target="#bd-toc-nav" data-offset="80">
    

    <div class="container-xl">
      <div class="row">
          
<div class="col-12 col-md-3 bd-sidebar site-navigation show" id="site-navigation">
    
        <div class="navbar-brand-box">
<a class="navbar-brand text-wrap" href="index.html">
  
  
  <h1 class="site-logo" id="site-title">genome-sampler documentation</h1>
  
</a>
</div>

<form class="bd-search d-flex align-items-center" action="search.html" method="get">
  <i class="icon fas fa-search"></i>
  <input type="search" class="form-control" name="q" id="search-input" placeholder="Search this book..." aria-label="Search this book..." autocomplete="off" >
</form>

<nav class="bd-links" id="bd-docs-nav" aria-label="Main navigation">
  <ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="intro.html">
   genome-sampler documentation
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Getting started
 </span>
</p>
<ul class="current nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="install.html">
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   installation instructions
  </a>
 </li>
 <li class="toctree-l1 current active">
  <a class="current reference internal" href="#">
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   usage tutorial
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Next steps
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="downstream-workflows.html">
   Downstream alignment and phylogenetics with QIIME 2
  </a>
 </li>
 <li class="toctree-l1">
  <a class="reference internal" href="other.html">
   Other topics
  </a>
 </li>
</ul>
<p class="caption">
 <span class="caption-text">
  Back matter
 </span>
</p>
<ul class="nav sidenav_l1">
 <li class="toctree-l1">
  <a class="reference internal" href="zreferences.html">
   List of works cited
  </a>
 </li>
</ul>

</nav>

 <!-- To handle the deprecated key -->

<div class="navbar_extra_footer">
  Powered by <a href="https://jupyterbook.org">Jupyter Book</a>
</div>

</div>


          


          
<main class="col py-md-3 pl-md-4 bd-content overflow-auto" role="main">
    
    <div class="row topbar fixed-top container-xl">
    <div class="col-12 col-md-3 bd-topbar-whitespace site-navigation show">
    </div>
    <div class="col pl-2 topbar-main">
        
        <button id="navbar-toggler" class="navbar-toggler ml-0" type="button" data-toggle="collapse"
            data-toggle="tooltip" data-placement="bottom" data-target=".site-navigation" aria-controls="navbar-menu"
            aria-expanded="true" aria-label="Toggle navigation" aria-controls="site-navigation"
            title="Toggle navigation" data-toggle="tooltip" data-placement="left">
            <i class="fas fa-bars"></i>
            <i class="fas fa-arrow-left"></i>
            <i class="fas fa-arrow-up"></i>
        </button>
        
        <div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn" aria-label="Download this page"><i
            class="fas fa-download"></i></button>

    
    <div class="dropdown-buttons">
        <!-- ipynb file if we had a myst markdown file -->
        
        <!-- Download raw file -->
        <a class="dropdown-buttons" href="_sources/tutorial.md"><button type="button"
                class="btn btn-secondary topbarbtn" title="Download source file" data-toggle="tooltip"
                data-placement="left">.md</button></a>
        <!-- Download PDF via print -->
        <button type="button" id="download-print" class="btn btn-secondary topbarbtn" title="Print to PDF"
            onClick="window.print()" data-toggle="tooltip" data-placement="left">.pdf</button>
    </div>
    
</div>
        <!-- Source interaction buttons -->

<div class="dropdown-buttons-trigger">
    <button id="dropdown-buttons-trigger" class="btn btn-secondary topbarbtn"
        aria-label="Connect with source repository"><i class="fab fa-github"></i></button>
    <div class="dropdown-buttons sourcebuttons">
        <a class="repository-button"
            href="https://github.com/caporaso-lab/genome-sampler"><button type="button" class="btn btn-secondary topbarbtn"
                data-toggle="tooltip" data-placement="left" title="Source repository"><i
                    class="fab fa-github"></i>repository</button></a>
        <a class="issues-button"
            href="https://github.com/caporaso-lab/genome-sampler/issues/new?title=Issue%20on%20page%20%2Ftutorial.html&body=Your%20issue%20content%20here."><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Open an issue"><i class="fas fa-lightbulb"></i>open issue</button></a>
        <a class="edit-button" href="https://github.com/caporaso-lab/genome-sampler/edit/master/docs/tutorial.md"><button
                type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip" data-placement="left"
                title="Edit this page"><i class="fas fa-pencil-alt"></i>suggest edit</button></a>
    </div>
</div>


        <!-- Full screen (wrap in <a> to have style consistency -->
        <a class="full-screen-button"><button type="button" class="btn btn-secondary topbarbtn" data-toggle="tooltip"
                data-placement="bottom" onclick="toggleFullScreen()" aria-label="Fullscreen mode"
                title="Fullscreen mode"><i
                    class="fas fa-expand"></i></button></a>

        <!-- Launch buttons -->

    </div>

    <!-- Table of contents -->
    <div class="d-none d-md-block col-md-2 bd-toc show">
        
        <div class="tocsection onthispage pt-5 pb-3">
            <i class="fas fa-list"></i> Contents
        </div>
        <nav id="bd-toc-nav">
            <ul class="nav section-nav flex-column">
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#download-tutorial-data">
   Download tutorial data
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#using-genome-sampler-snakemake-workflow">
   Using
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   (Snakemake workflow)
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#using-genome-sampler-step-by-step-instructions">
   Using
   <code class="docutils literal notranslate">
    <span class="pre">
     genome-sampler
    </span>
   </code>
   (step-by-step instructions)
  </a>
 </li>
 <li class="toc-h2 nav-item toc-entry">
  <a class="reference internal nav-link" href="#optional-alignment-and-phylogenetic-reconstruction-with-qiime-2">
   Optional: alignment and phylogenetic reconstruction with QIIME 2
  </a>
 </li>
</ul>

        </nav>
        
    </div>
</div>
    <div id="main-content" class="row">
        <div class="col-12 col-md-9 pl-md-3 pr-md-0">
        
              <div>
                
  <div class="section" id="genome-sampler-usage-tutorial">
<span id="usage-tutorial"></span><h1><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> usage tutorial<a class="headerlink" href="#genome-sampler-usage-tutorial" title="Permalink to this headline">¶</a></h1>
<p>This document provides illustrates how to use <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> on a small
tutorial data set.</p>
<div class="section" id="download-tutorial-data">
<h2>Download tutorial data<a class="headerlink" href="#download-tutorial-data" title="Permalink to this headline">¶</a></h2>
<p>The tutorial data set used here is intended for educational purposes only. If
you’re interested in using these sequences for other analyses, we recommend
starting with sequence repositories such as GISAID or NCBI Genbank, which may
have updated versions.</p>
<p>Download the tutorial sequences and corresponding metadata using the following
commands:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">mkdir</span> <span class="o">-</span><span class="n">p</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span> <span class="o">--</span><span class="n">output</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span> <span class="o">--</span><span class="n">output</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">focal</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">focal</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> <span class="o">--</span><span class="n">output</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span><span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span>
</pre></div>
</div>
</div>
<div class="section" id="using-genome-sampler-snakemake-workflow">
<h2>Using <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> (Snakemake workflow)<a class="headerlink" href="#using-genome-sampler-snakemake-workflow" title="Permalink to this headline">¶</a></h2>
<p>The full <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> workflow can be run using
<a class="reference external" href="https://snakemake.readthedocs.io/en/stable/">Snakemake</a>
<a class="bibtex reference internal" href="zreferences.html#snakemake-bioi" id="id1">[6]</a>. If you’d like to get started quickly and use default
parameters, start here.
If you’d like more control over your analysis or want to work through the steps
individually, move on to the next section.</p>
<p>Download the Snakemake and associated config file using <code class="docutils literal notranslate"><span class="pre">curl</span></code> as follows:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">Snakefile</span> <span class="o">--</span><span class="n">output</span> <span class="n">Snakefile</span>
<span class="n">curl</span> <span class="o">-</span><span class="n">sL</span> <span class="n">https</span><span class="p">:</span><span class="o">//</span><span class="n">raw</span><span class="o">.</span><span class="n">githubusercontent</span><span class="o">.</span><span class="n">com</span><span class="o">/</span><span class="n">caporaso</span><span class="o">-</span><span class="n">lab</span><span class="o">/</span><span class="n">genome</span><span class="o">-</span><span class="n">sampler</span><span class="o">/</span><span class="n">r2020</span><span class="o">.</span><span class="mi">8</span><span class="o">/</span><span class="n">snakemake</span><span class="o">/</span><span class="n">config</span><span class="o">.</span><span class="n">yaml</span> <span class="o">--</span><span class="n">output</span> <span class="n">config</span><span class="o">.</span><span class="n">yaml</span>
</pre></div>
</div>
<p>Place the resulting Snakemake file in the same folder as the sequence and
metadata files that you downloaded above. Then run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">snakemake</span>
</pre></div>
</div>
<p>When this workflow completes, there will be two primary output files that
you’ll use. <code class="docutils literal notranslate"><span class="pre">sequences.fasta</span></code> will contain your subsampled context sequences
and your focal sequences. You should use this file for downstream analyses,
such as alignment and phylogenetic analyses. <code class="docutils literal notranslate"><span class="pre">selection-summary.qzv</span></code> will
provide a summary of the sampling run. You can view this file using <a class="reference external" href="https://view.qiime2.org">QIIME 2
View</a>.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>After loading <code class="docutils literal notranslate"><span class="pre">selection-summary.qzv</span></code> with <a class="reference external" href="https://view.qiime2.org">QIIME 2
View</a>, click the <em>Provenance</em> tab to see full details
on the workflow that was executed. This allows you to review exactly what steps
were performed, what parameters were used for each step, and what versions of
<code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> and its dependencies were installed when you ran
<code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code>. Keep this file for your records so you can refer to it if
any of this information is needed in the future (for example, when publishing
your findings).</p>
</div>
<p>You should now be able to move on to analysis of your own data. See
<a class="reference internal" href="other.html#adapting-tutorial"><span class="std std-ref">Adapting Snakemake workflow for application to your own data</span></a> to learn about what changes you might want to make
to your <code class="docutils literal notranslate"><span class="pre">Snakefile</span></code> before running your own analysis.</p>
<p>🐍</p>
</div>
<div class="section" id="using-genome-sampler-step-by-step-instructions">
<h2>Using <code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> (step-by-step instructions)<a class="headerlink" href="#using-genome-sampler-step-by-step-instructions" title="Permalink to this headline">¶</a></h2>
<p>If you are working with the tutorial data above, then you will need to enter
the tutorial directory. If you are working with your own data, then replace
file paths as appropriate.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">cd</span> <span class="n">tutorial</span><span class="o">-</span><span class="n">data</span><span class="o">/</span>
</pre></div>
</div>
<p>You’ll begin the workflow by importing the fasta files into QIIME 2
Artifacts. QIIME 2 Artifacts are structured zip files which will contain the
data (still in fasta format), but also some QIIME 2-specific information that
(among other things) automates your method recording so you don’t have to do
that.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">tools</span> <span class="kn">import</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">path</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">path</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">format</span> <span class="n">GISAIDDNAFASTAFormat</span> \
  <span class="o">--</span><span class="nb">type</span> <span class="n">FeatureData</span><span class="p">[</span><span class="n">Sequence</span><span class="p">]</span>

<span class="n">qiime</span> <span class="n">tools</span> <span class="kn">import</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">path</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">fasta</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">path</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="nb">format</span> <span class="n">GISAIDDNAFASTAFormat</span> \
  <span class="o">--</span><span class="nb">type</span> <span class="n">FeatureData</span><span class="p">[</span><span class="n">Sequence</span><span class="p">]</span>
</pre></div>
</div>
<p>If you’re obtaining context sequences from a public repository, you may
encounter context sequences that don’t have associated metadata records. Steps
in this workflow that require context sequence metadata would fail as a
result. At this stage we therefore filter any context sequences that don’t
have metadata records.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">data</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Next, we’ll apply a quality filter to the sequence data. Technically this is
an optional step for both the context and focal sequences, but in practice if
you obtain your context sequences from a public repository you should
consider the context sequence filtering to be essential. You can use this to
remove sequences based on their length (if they’re too short or too long),
and based on the proportion of ambiguous (e.g., <code class="docutils literal notranslate"><span class="pre">N</span></code>) nucleotide characters
that they contain. Here we’ll apply this both to the context sequences and
the focal sequences. We’ll remove sequences with a proportion of 1% or more
ambiguous characters.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">sequences</span> <span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">-</span><span class="n">w</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="nb">max</span><span class="o">-</span><span class="n">proportion</span><span class="o">-</span><span class="n">ambiguous</span> <span class="mf">0.01</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">sequences</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">sequences</span> <span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="nb">max</span><span class="o">-</span><span class="n">proportion</span><span class="o">-</span><span class="n">ambiguous</span> <span class="mf">0.01</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">sequences</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>If you’d like to see what other options you can control during this filtering
step, you can run:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> <span class="o">--</span><span class="n">help</span>
</pre></div>
</div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>The <code class="docutils literal notranslate"><span class="pre">--help</span></code> parameter can be provided to any of the commands that are used
in this tutorial.</p>
</div>
<p>At any time, you could get some summary information about your sequences
using the following command:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">tabulate</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qzv</span>
</pre></div>
</div>
<p>That command will create a QIIME 2 visualization that you can view using
<a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>. Try viewing that file and finding
information such as the number of sequences present in this file and the
median length of the sequences. (Your data is not uploaded to a server when
you visit <a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>, so you don’t need to be
conerned about exposing sensitive research data.)</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>If some of your focal sequences are present in your context sequence
collection (for example because you submitted your sequences to GISAID before
downloading GISAID), you should remove those sequences from either your focal
or context sequence collection. See <a class="reference internal" href="other.html#removing-sequences"><span class="std std-ref">Removing sequences present in both focal and context sequence collections</span></a> for instructions
on how to do that.</p>
</div>
<p>We’re now ready to start sampling our data, and we’ll do this in three steps.
These subsampling steps are independent, so can be run in any order.</p>
<p>First, we’ll sample across time. By default, this will select seven genomes
per seven day period. The file that gets generated as a result here isn’t
something that is very useful to view directly, but we’ll use it in a few
minutes to see how many sequences were retained at this step.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">longitudinal</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">dates</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">dates</span><span class="o">-</span><span class="n">column</span> <span class="n">date</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">date</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Next, we’ll sample across viral diversity. This will cluster sequences at a
percent identity threshold of 99.95%, and select the centroid sequence from
each cluster to include in downstream analyses.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">diversity</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">percent</span><span class="o">-</span><span class="nb">id</span> <span class="mf">0.9995</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Last, we’ll sample near neighbors of the focal sequences from the context
sequences. Of the near-neighbor sequences that are identified for each focal
sequence (up to 10 by default), 3 sequences will be selected at random such
that each <code class="docutils literal notranslate"><span class="pre">location</span></code> (defined in the <code class="docutils literal notranslate"><span class="pre">context-metadata.tsv</span></code> file) has an
equal probability of being selected for inclusion in downstream analysis.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">sample</span><span class="o">-</span><span class="n">neighbors</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">locale</span><span class="o">-</span><span class="n">file</span> <span class="n">context</span><span class="o">-</span><span class="n">metadata</span><span class="o">.</span><span class="n">tsv</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">locale</span><span class="o">-</span><span class="n">column</span> <span class="n">location</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">percent</span><span class="o">-</span><span class="nb">id</span> <span class="mf">0.9999</span> \
  <span class="o">--</span><span class="n">p</span><span class="o">-</span><span class="n">samples</span><span class="o">-</span><span class="n">per</span><span class="o">-</span><span class="n">cluster</span> <span class="mi">3</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">selection</span> <span class="n">neighbor</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Now, we’ll combine the results of the three sampling approaches and generate
a summary of the full selection process.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">combine</span><span class="o">-</span><span class="n">selections</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">selections</span> <span class="n">date</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">neighbor</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">combined</span><span class="o">-</span><span class="n">selection</span> <span class="n">master</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">genome</span><span class="o">-</span><span class="n">sampler</span> <span class="n">summarize</span><span class="o">-</span><span class="n">selections</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">selections</span> <span class="n">date</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">diversity</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> <span class="n">neighbor</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">visualization</span> <span class="n">selection</span><span class="o">-</span><span class="n">summary</span><span class="o">.</span><span class="n">qzv</span>
</pre></div>
</div>
<p>The <code class="docutils literal notranslate"><span class="pre">selection-summary.qzv</span></code> file provides a summary of the sampling run. You
can view this file using <a class="reference external" href="https://view.qiime2.org">QIIME 2 View</a>. Be sure to
click the <em>Provenance</em> tab after loading the file on that page - that will
provide full details on the workflow that was executed. How many sequences
were retained by each sampling step?</p>
<p>We’re now ready to start compiling our final data set. To do this, we’ll use
the <code class="docutils literal notranslate"><span class="pre">master-selection.qza</span></code> file to select specific context sequences from the
<code class="docutils literal notranslate"><span class="pre">filtered-context-seqs.qza</span></code> file that was generated earlier.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="nb">filter</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">filtered</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">m</span><span class="o">-</span><span class="n">metadata</span><span class="o">-</span><span class="n">file</span> <span class="n">master</span><span class="o">-</span><span class="n">selection</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">filtered</span><span class="o">-</span><span class="n">data</span> <span class="n">subsampled</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span>
</pre></div>
</div>
<p>Next, merge the resulting subsampled context sequences with the focal
sequences to create the final QIIME 2 artifact. Then, export that to generate
a fasta file that you can use in downstream analysis.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">qiime</span> <span class="n">feature</span><span class="o">-</span><span class="n">table</span> <span class="n">merge</span><span class="o">-</span><span class="n">seqs</span> \
  <span class="o">--</span><span class="n">i</span><span class="o">-</span><span class="n">data</span> <span class="n">subsampled</span><span class="o">-</span><span class="n">context</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> <span class="n">filtered</span><span class="o">-</span><span class="n">focal</span><span class="o">-</span><span class="n">seqs</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">o</span><span class="o">-</span><span class="n">merged</span><span class="o">-</span><span class="n">data</span> <span class="n">sequences</span><span class="o">.</span><span class="n">qza</span>

<span class="n">qiime</span> <span class="n">tools</span> <span class="n">export</span> \
  <span class="o">--</span><span class="nb">input</span><span class="o">-</span><span class="n">path</span> <span class="n">sequences</span><span class="o">.</span><span class="n">qza</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="n">path</span> <span class="n">sequences</span><span class="o">.</span><span class="n">fasta</span> \
  <span class="o">--</span><span class="n">output</span><span class="o">-</span><span class="nb">format</span> <span class="n">DNAFASTAFormat</span>
</pre></div>
</div>
</div>
<div class="section" id="optional-alignment-and-phylogenetic-reconstruction-with-qiime-2">
<h2>Optional: alignment and phylogenetic reconstruction with QIIME 2<a class="headerlink" href="#optional-alignment-and-phylogenetic-reconstruction-with-qiime-2" title="Permalink to this headline">¶</a></h2>
<p>QIIME 2 contains tools for preliminary sequence alignment and
phylogenetic tree generation in the
<a class="reference external" href="https://docs.qiime2.org/2020.2/plugins/available/alignment/">q2-alignment</a>
and
<a class="reference external" href="https://docs.qiime2.org/2020.2/plugins/available/phylogeny/">q2-phylogeny</a>
plugins. QIIME 2 is not designed for high-accuracy phylogenetic analysis, but
it can help you to generate and visualize initial alignments and trees.</p>
<p>These steps would take the <code class="docutils literal notranslate"><span class="pre">sequences.qza</span></code> file as input, so if you’re
interested in using these you could postpone or skip the export step that you
ran above.</p>
<p>If you’d like to learn more about this, refer to <a class="reference internal" href="downstream-workflows.html#downstream"><span class="std std-ref">Downstream alignment and phylogenetics with QIIME 2</span></a>.</p>
</div>
</div>

    <script type="text/x-thebe-config">
    {
        requestKernel: true,
        binderOptions: {
            repo: "binder-examples/jupyter-stacks-datascience",
            ref: "master",
        },
        codeMirrorConfig: {
            theme: "abcdef",
            mode: "python"
        },
        kernelOptions: {
            kernelName: "python3",
            path: "./."
        },
        predefinedOutput: true
    }
    </script>
    <script>kernelName = 'python3'</script>

              </div>
              
        </div>
    </div>
    
    
    <div class='prev-next-bottom'>
        
    <a class='left-prev' id="prev-link" href="install.html" title="previous page"><code class="docutils literal notranslate"><span class="pre">genome-sampler</span></code> installation instructions</a>
    <a class='right-next' id="next-link" href="downstream-workflows.html" title="next page">Downstream alignment and phylogenetics with QIIME 2</a>

    </div>
    <footer class="footer mt-5 mt-md-0">
    <div class="container">
      <p>
        
          By The genome-sampler developers<br/>
        
            &copy; Copyright 2020.<br/>
      </p>
    </div>
  </footer>
</main>


      </div>
    </div>

    
  <script src="_static/js/index.3da636dd464baa7582d2.js"></script>


    
    <!-- Google Analytics -->
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-86713402-2', 'auto');
      ga('set', 'anonymizeIp', true);
      ga('send', 'pageview');
    </script>
    <script async src='https://www.google-analytics.com/analytics.js'></script>
    <!-- End Google Analytics -->
    
  </body>
</html>